source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/multi-select/index.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { ResourceObject } from 'jsonapi-typescript';
+
+import { attributesFor, relationshipFor } from '@data';
+import { i18nProps } from '@lib/i18n';
+import { Checkbox } from 'semantic-ui-react';
+import ProductIcon from '@ui/components/product-icon';
+
+import { isEmpty } from '@lib/collection';
+
+interface IOwnProps<T> {
+  className?: string;
+  name: string;
+  list: T[];
+  selected: T[];
+  onChange: (el: T) => void;
+}
+
+export type IProps<T extends ResourceObject> =
+  & IOwnProps<T>
+  & i18nProps;
+
+export class MultiSelect extends React.Component<IProps<ResourceObject>> {
```

Comment:
Here is where is was complaining if just use

`export class MultiSelect extends React.Component<IProps<T>>` or 
`export class MultiSelect extends React.Component<IProps>`

Reply:
What if you do:

`export class MultiSelect<T> extends React.Component React.Component<IProps<T>> { ...}`
?

Reply:
This one works better

`export type IProps<T> =
  & IOwnProps<T>
  & i18nProps;

export class MultiSelect<T extends ResourceObject> extends React.Component<IProps<T>> {`